client/src/Components/AverageRating.jsx

```diff
@@ -1,9 +1,112 @@
 import React from 'react';
+import styled from 'styled-components';
 
-const AverageRating = ({ average, ratings }) => (
-  <div>
-    Hello from AverageRating component!
-  </div>
-);
+/* -------------------------  STYLES ------------------------- */
+
+const DivStarRatingsContainer = styled.div`
+  margin: 5px 0 19px;
+  box-sizing: border-box;
+  inline-size: 150px;
+`;
+
+const DivStarRatingsSubContainer = styled.a`
+  display: flex;
+  align-items: center;
+  text-decoration: none;
+  cursor: pointer;
+`;
+
+const SpanNumericalRating = styled.span`
+  font-size: 2.2rem;
+  font-weight: 600;
+  line-height: 1;
+`;
+
+const SpanStarRatingsWrapper = styled.span`
+  flex-direction: column;
+  margin: 2px 0 0 5px;
+  width: 80px;
+`;
+
+const UlStarRating = styled.ul`
+  margin: 0 0 2px;
+  list-style: none;
+  padding: 0;
+  font-size: 12px;
+  float: left;
+  height: 13.4px;
+`;
+
+const SpanStarText = styled.span`
+  font-weight: 400;
+  line-height: 18.2px;
+  font-size: 14px;
+  line-height: normal;
+`;
+
+const IStar = styled.i`
+  font-size: 15px;
+  font-style: normal;
+  color: ${({ starColor }) => starColor ? "#ffc120" : "#e6e7e8;"};
+  font-family: Material Icons;
+`;
+
+/* -------------------------  RENDER ------------------------- */
+
+const AverageRating = ({ average, ratings }) => {
+
+  // Generate colored stars based on passed in average
+  const stars = [];
+  let iconString = '';
+  let starColor;
+  let remaining = average;
+
+  // For each star, loop through and generate a jsx element with a corresponding star color and shape
+  for (let i = 0; i < 5; i += 1) {
+
+    // Determine star shape (filled or filled half)
+    iconString = (remaining >= 1 || (remaining > 0 && remaining.toFixed(1) % 1 >= 0.8))
```

Comment:
For all of the Number.toFixed() calls. What is the shape of the average rating when it gets to this point. If it is already at one decimal place, this would be duplicate work. If it isn't there is a lot of duplicate code. Maybe consider "toFixing" the average then make all of the comparisons.

Reply:
I was having an issue with the JS interpreter doing something like:
`2.3 - 1 => 1.22222222222222222298`
😂